model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TypeInfoTests.xtend

```diff
@@ -14,1049 +15,1552 @@ import hu.blackbelt.judo.meta.jsl.jsldsl.EntityDerivedDeclaration
-		assertEquals(TypeInfo.PrimitiveType.NUMERIC, TypeInfo.getTargetType(asMillisecondsField.expression).getPrimitive);
-		assertEquals(TypeInfo.PrimitiveType.TIMESTAMP, TypeInfo.getTargetType(fromMillisecondsField.expression).getPrimitive);
-		assertEquals(TypeInfo.PrimitiveType.TIMESTAMP, TypeInfo.getTargetType(timestampField.expression).getPrimitive);
-		assertEquals(TypeInfo.PrimitiveType.TIMESTAMP, TypeInfo.getTargetType(plusField.expression).getPrimitive);
-
-	}
-
-	@Test
-	def void testEntityInstanceFunctions() {
-		val m = '''
-			model TestModel;
-			
-			type boolean Boolean;
-
-			entity T1 {
-			}
-
-			entity T2 extends T1 {
-			}
-
-			entity Test {
-				field T1 t1;
-				field T2 t2;
-				field T2[] t2s;
-
-				derived Boolean typeOf => self.t1!typeOf(entityType = T1);
-				derived Boolean kindOf => self.t1!kindOf(entityType = T1);
-				derived Test container => self.t1!container(entityType = Test);
-				derived T1 asType => self.t2!asType(entityType = T1);
-				derived Boolean memberOf => self.t2!memberOf(instances = self.t2s);
-
-			}
-		'''.parse.fromModel
-	
-		val testEntity = m.entityByName("Test") 
-		val t1Field = testEntity.memberByName("t1") as EntityFieldDeclaration		
-
-		val typeOfField = testEntity.memberByName("typeOf") as EntityDerivedDeclaration
-		val kindOfField = testEntity.memberByName("kindOf") as EntityDerivedDeclaration
-		val containerField = testEntity.memberByName("container") as EntityDerivedDeclaration
-		val asTypeField = testEntity.memberByName("asType") as EntityDerivedDeclaration
-
-		assertEquals(TypeInfo.PrimitiveType.BOOLEAN, TypeInfo.getTargetType(typeOfField.expression).getPrimitive)
-		assertEquals(TypeInfo.PrimitiveType.BOOLEAN, TypeInfo.getTargetType(kindOfField.expression).getPrimitive)
-		
-		val TypeInfo containerTypeInfo = TypeInfo.getTargetType(containerField.expression)
-		assertEquals(testEntity, containerTypeInfo.getEntity)
-		assertEquals(false, containerTypeInfo.isCollection)		
-
-		val TypeInfo asTypeTypeInfo = TypeInfo.getTargetType(asTypeField.expression)
-		assertEquals(t1Field.referenceType, asTypeTypeInfo.getEntity)
-		assertEquals(false, containerTypeInfo.isCollection)		
-
-		val memberOfField = testEntity.memberByName("memberOf") as EntityDerivedDeclaration
-		assertEquals(TypeInfo.PrimitiveType.BOOLEAN, TypeInfo.getTargetType(memberOfField.expression).getPrimitive)
-
-	}
-
-	@Test
-	def void testSelectorFunctions() {
-		val m = '''
-			model TestModel;
-			
-			type string String(min-size = 0, max-size = 32);
-
-			entity T1 {
-				field String name;
-			}
-
-			entity Test {
-				field T1[] t1s;
-				field T1 t1;
-
-				derived T1[] head => self.t1s!first(t | t.name);
-				derived T1[] tail => self.t1s!last(t | t.name);
-				derived T1 any => self.t1s!any();
-				derived Integer size => self.t1s!size();
-				derived T1[] asCollection => self.t1!asCollection(entityType = T1);
```

Comment:
If the only reason to remove this was only because `t1` is not a collection, this could have been fixed by using `ts1` instead of `ts` thus a test for "all" collection function would remain here.

Reply:
OK. I will add it again and it will use the t1s field.